src/feat/pitch-functions.cc

```diff
@@ -835,19 +844,35 @@ void OnlinePitchFeatureImpl::ExtractFrame(
   int32 offset = static_cast<int32>(sample_index -
                                     downsampled_samples_processed_);
 
+  // Treat edge cases first
+  if (sample_index < 0) {
+    // Part of the frame is before the beginning of the signal. This
+    // should only happen if opts_.snip_edges == false, when we are
+    // processing the first few frames of signal. In this case
+    // we pad with zeros.
+    KALDI_ASSERT(opts_.snip_edges == false);
```

Comment:
Hi Dan,

I have just retested on 10000 wav files of various lengths with the default settings apart from the snip-edges option.
 - snip-edges=false modifies the output of every single file, but the number of output frames is identical
 - snip-edges=true does not modify the output of any file.

So I would say the change should be safe, unless some of the recipes were using snip-edges=false already; but even then I would suspect it would improve their results.